### Codebook

Voted last election (vote): 1 = Yes, 2 = No, 3 = Not eligible to vote, 7 = Refusal, 8 = Don't know, 7 = No answer. 

Internet use (netusoft): 1 = Never , 2 = Only occasionally , 3 = A few times a week, 4 = Most days, 5 = Every day, 7 = refusal,                         8 = don't know, 7 = no answer. 


In [1]:
import os
import pandas as pd
import researchpy as rp
import numpy as np

In [2]:
os.getcwd()

'C:\\Users\\danin'

In [3]:
path = 'C:\\Users\\danin\\Desktop\\Datasets\\'

In [4]:
df = pd.read_csv(path +'ESS10.csv')

C:\Users\danin\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3444: DtypeWarning: Columns (159,165,167,578) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [5]:
df.head(2)

name  essround  edition    proddate   idno cntry   dweight   pspwght  \
0  ESS10e02_2        10      2.2  21.12.2022  10002    BG  1.939384  1.290706   
1  ESS10e02_2        10      2.2  21.12.2022  10006    BG  1.651595  1.430878   

    pweight  anweight  ...                hinwe                iinwe  \
0  0.217716  0.281008  ...  2021-07-24 13:11:27  2021-07-24 13:11:35   
1  0.217716  0.311526  ...  2021-08-28 08:07:22  2021-08-28 08:07:39   

                 kinwe                vinwe                inwde  \
0  2021-07-24 13:12:20  2021-07-24 13:12:21  2021-07-24 13:13:01   
1  2021-08-28 08:14:39  2021-08-28 08:14:43  2021-08-28 08:36:39   

                 jinws                jinwe  inwtm  mode  domain  
0  2021-07-24 13:12:33  2021-07-24 13:13:06   53.0     1     NaN  
1  2021-08-28 08:25:43  2021-08-28 08:36:49   36.0     1     NaN  

[2 rows x 586 columns]

In [6]:
df.tail(2)

name  essround  edition    proddate   idno cntry   dweight  \
33349  ESS10e02_2        10      2.2  21.12.2022  27846    SK  0.854279   
33350  ESS10e02_2        10      2.2  21.12.2022  27858    SK  0.702292   

        pspwght  pweight  anweight  ...                hinwe  \
33349  0.624287   0.3238  0.202144  ...  2021-07-21 14:06:56   
33350  0.581970   0.3238  0.188442  ...  2021-07-11 10:37:13   

                     iinwe                kinwe                vinwe  \
33349  2021-07-21 14:08:48  2021-07-21 14:13:06  2021-07-21 14:14:41   
33350  2021-07-11 10:40:30  2021-07-11 10:45:47  2021-07-11 10:48:52   

                     inwde                jinws                jinwe  inwtm  \
33349  2021-07-21 14:17:31  2021-07-21 14:16:38  2021-07-21 14:18:38   43.0   
33350  2021-07-11 10:52:58  2021-07-11 10:52:10  2021-07-11 10:54:10   49.0   

       mode  domain  
33349     1     1.0  
33350     1     1.0  

[2 rows x 586 columns]

In [7]:
#Subsetting only the Dutch cases, vote, and internet use
ess_nl=df[['vote','netusoft','cntry']][df.cntry=='NL']

In [8]:
ess_nl.head(4)

vote  netusoft cntry
25962     1         5    NL
25963     1         5    NL
25964     1         5    NL
25965     2         5    NL

In [9]:
ess_nl.tail(4)

vote  netusoft cntry
27428     1         5    NL
27429     2         5    NL
27430     2         5    NL
27431     3         5    NL

In [10]:
ess_nl.shape
#1470 respondents

(1470, 3)

In [11]:
rp.codebook(ess_nl)

Variable: vote    Data Type: int64 

 Number of Obs.: 1470 
 Number of missing obs.: 0 
 Percent missing: 0.0 
 Number of unique values: 4 

 Range: [1, 7] 
 Mean: 1.2 
 Standard Deviation: 0.55 
 Mode: 1 
 10th Percentile: 1.0 
 25th Percentile: 1.0 
 50th Percentile: 1.0 
 75th Percentile: 1.0 
 90th Percentile: 2.0 





Variable: netusoft    Data Type: int64 

 Number of Obs.: 1470 
 Number of missing obs.: 0 
 Percent missing: 0.0 
 Number of unique values: 5 

 Range: [1, 5] 
 Mean: 4.78 
 Standard Deviation: 0.74 
 Mode: 5 
 10th Percentile: 4.0 
 25th Percentile: 5.0 
 50th Percentile: 5.0 
 75th Percentile: 5.0 
 90th Percentile: 5.0 





Variable: cntry    Data Type: object 

 Number of Obs.: 1470 
 Number of missing obs.: 0 
 Percent missing: 0.0 
 Number of unique values: 1 

 Data Values and Counts: 
 
 Values  Frequency
    NL       1470






In [12]:
#summary stats for vote
ess_nl['vote'].describe()

count    1470.000000
mean        1.202721
std         0.549515
min         1.000000
25%         1.000000
50%         1.000000
75%         1.000000
max         7.000000
Name: vote, dtype: float64

In [13]:
#Internet use
ess_nl['netusoft'].describe()

count    1470.000000
mean        4.783673
std         0.744944
min         1.000000
25%         5.000000
50%         5.000000
75%         5.000000
max         5.000000
Name: netusoft, dtype: float64

In [14]:
ess_nl['vote'].value_counts().sort_index()

1    1255
2     140
3      73
7       2
Name: vote, dtype: int64

In [15]:
ess_nl['netusoft'].value_counts().sort_index()

1      33
2      20
3      25
4      76
5    1316
Name: netusoft, dtype: int64

In [16]:
#Recode categorical variables. 
dictionary =  { 'vote': {1:'yes', 2:'no', 3: 'other' , 7: 'other'},  #Not eligible and Refusal to be set as missing
               'netusoft':  { 1: 'never', 2: 'only occasionally', 3: 'a few times a week',
                             4: 'most days', 5: 'every day'}}
                  

In [17]:
#Replace with the created dictionary
ess_nl.replace(dictionary, inplace=True)

In [18]:
ess_nl.vote[ess_nl.vote == 'other'] = np.nan

In [19]:
#check missings
ess_nl.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1470 entries, 25962 to 27431
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   vote      1395 non-null   object
 1   netusoft  1470 non-null   object
 2   cntry     1470 non-null   object
dtypes: object(3)
memory usage: 45.9+ KB


In [20]:
ess_nl['vote'].isnull().sum()

75

## Chi square test and Cramer's V

In [21]:
#Convert to categorical
ess_nl['vote']=ess_nl['vote'].astype('category')

In [22]:
#Convert to categorical
ess_nl["netusoft"] = ess_nl["netusoft"].astype('category')

#### Cross tab with column percentages


In [23]:
crosstab = rp.crosstab(ess_nl['vote'], ess_nl['netusoft'], prop = "col")
crosstab

netusoft                                                \
netusoft a few times a week every day most days   never only occasionally   
vote                                                                        
no                     16.0      9.49      8.11   27.27              15.0   
yes                    84.0     90.51     91.89   72.73              85.0   
All                   100.0    100.00    100.00  100.00             100.0   

                  
netusoft     All  
vote              
no         10.04  
yes        89.96  
All       100.00

There appears to be an association: more frequent the internet use, more likely to vote.

In [27]:
obs_counts, chi2, exp_counts = rp.crosstab(ess_nl['vote'], ess_nl['netusoft'], test = 'chi-square', expected_freqs = True,
                                           margins=False)

In [28]:
#Observed frequencies
obs_counts

netusoft                                            
netusoft a few times a week every day most days never only occasionally
vote                                                                   
no                        4       118         6     9                 3
yes                      21      1125        68    24                17

In [29]:
#Expected frequencies
exp_counts

netusoft                                    \
netusoft a few times a week   every day  most days      never   
vote                                                            
no                 2.508961   124.74552   7.426523   3.311828   
yes               22.491039  1118.25448  66.573477  29.688172   

                            
netusoft only occasionally  
vote                        
no                2.007168  
yes              17.992832

Some expected frequencies are < 5. 

In [30]:
chi2

Chi-square test  results
0  Pearson Chi-square ( 4.0) =   13.1003
1                    p-value =    0.0108
2                 Cramer's V =    0.0969

Statistically significant association between vote and internet use frequency, Chi2(4) = 13.10, p <0.05. 
The strength is, however, weak Cramer's V = .10.

In [31]:
ess_nl.to_csv('working_data.csv', index=False)